src/ripple/rpc/handlers/AccountObjects.cpp

```diff
@@ -72,12 +72,49 @@ Json::Value doAccountObjects (RPC::Context& context)
     if (! ledger->exists(keylet::account (accountID)))
         return rpcError (rpcACT_NOT_FOUND);
 
-    auto [rpcStatus, type] = RPC::chooseLedgerEntryType(params);
-    if (rpcStatus)
+    std::optional<std::vector<LedgerEntryType>> typeFilter;
+
+    if (params.isMember(jss::deletion_blockers_only) &&
+        params[jss::deletion_blockers_only].asBool())
     {
-        result.clear();
-        rpcStatus.inject(result);
-        return result;
+        struct {
+            Json::StaticString name;
+            LedgerEntryType type;
+        } static constexpr deletionBlockers[] = {
+            { jss::check,           ltCHECK },
+            { jss::escrow,          ltESCROW },
+            { jss::payment_channel, ltPAYCHAN },
+            { jss::state,           ltRIPPLE_STATE }
+        };
```

Comment:
Heh. I kinda like that. But I think I'd just give the struct a name, otherwise it'll take a minute for people to decipher the code, since we rarely see that construct.

Reply:
I think you misunderstood me. I'm objecting to putting the declaration of the struct with the declaration of the array. _Just_ adding a typename doesn't fix that. We also need to use that typename in the declaration.

Looking more closely, I'd probably just use `std::pair` here; we don't actually use the names in the struct, so something like:
```
        static constexpr std::pair<Json::StaticString, LedgerEntryType>
            deletionBlockers[] = {{jss::check, ltCHECK},
                                  {jss::escrow, ltESCROW},
                                  {jss::payment_channel, ltPAYCHAN},
                                  {jss::state, ltRIPPLE_STATE}};
```

If you really want to keep the struct, I'd do something like:
```
        static constexpr NameType deletionBlockers[] = {
            {jss::check, ltCHECK},
            {jss::escrow, ltESCROW},
            {jss::payment_channel, ltPAYCHAN},
            {jss::state, ltRIPPLE_STATE}};
```
I prefer `std::pair` as we don't actually make use of the struct, but I don't object to the struct if you really want to keep it.

Reply:
Oh, I'd rather keep the pattern as-is then (no name for the struct). It has advantages of not polluting any namespaces, and can't be mistakenly re-used for the wrong purpose (you might be able to `decltype(deletionBlockers[0])`, but only inside the same function. I think that `std::pair` is less readable. With the first iteration I know the first parameter is meant to be the "name" for whatever this entry is, `std::pair` loses that bit of information, I just know that it's a `Json::StaticString`. 

I think `std::pair/tuple` are great for generic programming where you don't know what the names should be. But we already have all the information, so I'd rather just use a struct.

Reply:
I'm fine with keeping the original.

Reply:
I find `struct` superior to `std::pair` 99.9% of the time. For me, this happens to be the 0.1% of the time when it’s the other way around.

We never use the names of the `struct` directly and the code is very localized—all within 10-15 lines.

I’d much prefer if we did: `std::pair<Json::StaticString, LedgerEntryType> = { { blah, blah }, ... }`.

Rationale: the nameless struct declaration is cool but “exotic” enough that people will do a double take; the names introduced by the `struct` and the strong typing don’t help, especially given the limited scope.

I’m not speaking _ex cathedra_ and I’m not gonna hold this up over this; just my personal opinion.